DDG4/edm4hep/EDM4hepFileReader.cpp

```diff
@@ -190,63 +190,64 @@ namespace dd4hep::sim {
     m_currEvent = event_number;
     podio::Frame frame = m_reader.readFrame("events", event_number);
     const auto& primaries = frame.get<edm4hep::MCParticleCollection>(m_collectionName);
+    const auto availableCollections = frame.getAvailableCollections();
     int eventNumber = event_number, runNumber = 0;
 #if PODIO_BUILD_VERSION >= PODIO_VERSION(1, 6, 0)
     if (primaries.hasID()) {
 #else
     if (primaries.isValid()) {
 #endif
       //Read the event header collection and add it to the context as an extension
-      const auto& eventHeaderCollection = frame.get<edm4hep::EventHeaderCollection>(m_eventHeaderCollectionName);
-      if(eventHeaderCollection.size() == 1){
-        const auto& eh = eventHeaderCollection.at(0);
-        eventNumber = eh.getEventNumber();
-        runNumber = eh.getRunNumber();
-        try {
-          Geant4Context* ctx = context();
-          ctx->event().addExtension<edm4hep::MutableEventHeader>(new edm4hep::MutableEventHeader(eh.clone()));
-        } catch(std::exception& ) {}
-        // Create input event parameters context
-        try {
-          Geant4Context* ctx = context();
-          EventParameters *parameters = new EventParameters();
-          parameters->setRunNumber(runNumber);
-          parameters->setEventNumber(eventNumber);
-          parameters->ingestParameters(frame.getParameters());
-          ctx->event().addExtension<EventParameters>(parameters);
-        } catch(std::exception &) {}
-      } else {
-        printout(WARNING,"EDM4hepFileReader","No EventHeader collection found or too many event headers!");
-        try {
-          Geant4Context* ctx = context();
-          EventParameters *parameters = new EventParameters();
-          parameters->setRunNumber(0);
-          parameters->setEventNumber(event_number);
-          parameters->ingestParameters(frame.getParameters());
-          ctx->event().addExtension<EventParameters>(parameters);
-        } catch(std::exception &) {}
+      if (std::find(availableCollections.begin(), availableCollections.end(), m_eventHeaderCollectionName) !=
+          frame.getAvailableCollections().end()) {
```

Comment:
```suggestion
          availableCollections.end()) {
```
?

Reply:
Well, since `getAvailableCollection()` creates a new vector every time it is called `getAvailableCollections().end() != getAvailableCollections().end()`, which makes for some hilarious debugging.